routes/items/[id].tsx

```diff
@@ -91,23 +88,23 @@ export default async function ItemsItemPage(
   _req: Request,
   ctx: RouteContext<undefined, SignedInState>,
 ) {
-  const { id } = ctx.params;
-  const item = await getItem(id);
+  const itemId = ctx.params.id;
+  const item = await getItem(itemId);
   if (item === null) return await ctx.renderNotFound();
 
-  const pageNum = calcPageNum(ctx.url);
-  const allComments = await getCommentsByItem(id);
-  const comments = allComments
-    .sort((a, b) => b.createdAt.getTime() - a.createdAt.getTime())
-    .slice((pageNum - 1) * PAGE_LENGTH, pageNum * PAGE_LENGTH);
+  const cursor = ctx.url.searchParams.get("cursor") ?? undefined;
+  const iter = listCommentsByItem(itemId, { cursor });
+  const comments = await valuesFromIter(iter);
+
+  /** @todo https://github.com/denoland/deno/issues/20173 */
+  const nextPageIter = listCommentsByItem(itemId, { cursor: iter.cursor });
```

Comment:
Do not do a second list. Instead, increase the limit by one in the first `listCommentsByItem` call and drop the last entry.

Reply:
Done. Is this solution optimal? WDYT?